Applications/Spire/Source/Ui/DestinationFilterPanel.cpp

```diff
@@ -0,0 +1,105 @@
+#include "Spire/Ui/DestinationFilterPanel.hpp"
+#include "Spire/Ui/AnyInputBox.hpp"
+#include "Spire/Ui/CustomQtVariants.hpp"
+#include "Spire/Ui/DestinationBox.hpp"
+
+using namespace boost::signals2;
+using namespace Nexus;
+using namespace Spire;
+
+struct DestinationFilterQueryModel : ComboBox::QueryModel {
```

Comment:
Avoid repetition, if you find yourself copying/pasting code, consider finding a way of implementing a solution that reduces the amount of copying/pasting.

Reply:
I proposed the `ExcludingQueryModel` to avoid implementing two similar models which are used for the `SecurityFIlterModel` and `DestinationFilterModel`. The model accepts a function to convert `std::any` to `QString`, which can be customized by the `SecurityFilterModel` and `DestinationFilterModel`.

https://gist.github.com/shaobinspire/1cffead77a60807b26a534d2665d90d1

Reply:
What is the purpose of the the ```Converter```?

Reply:
The `Converter` is used to convert `std::any` to a `QString` based on the specific type. For example, for the DestinationFilterPanel, the `Converter` casts the `Destination` to `QString`:
```
  auto box = new DestinationBox(std::make_shared<ExcludingQueryModel>(model, matches,
      [] (const std::any& value) {
        return displayTextAny(std::any_cast<const DestinationDatabase::Entry&>(value).m_id);
      }));
```

For the SecurityFilterPanel, the `Converter` casts the `Security` to `QString`:
```
  auto box = new SecurityBox(std::make_shared<ExcludingQueryModel>(model, matches,
      [] (const std::any& value) {
        return displayTextAny(std::any_cast<SecurityInfo&>(value).m_security);
      }));
```

Reply:
That tells me what it does, it does not tell me its purpose.

Reply:
I don't follow the API. ```ExcludingQueryModel``` is now a template parameterized over a type ```T``` but ```T``` is not used anywhere. Think carefully about the API and then feel free to propose it on Asana.